propagator/jaeger/lib/opentelemetry/propagator/jaeger/text_map_injector.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'cgi'
+
+module OpenTelemetry
+  # Namespace for OpenTelemetry propagator extension libraries
+  module Propagator
+    # Namespace for OpenTelemetry Jaeger propagation
+    module Jaeger
+      # Injects context into carriers
+      class TextMapInjector
+        # Returns a new TextMapInjector that extracts Jaeger context using the
+        # specified header keys
+        #
+        # @param [optional Setter] default_setter The default setter used to
+        #   write context into a carrier during inject. Defaults to a
+        #   {OpenTelemetry::Context:Propagation::TextMapSetter} instance.
+        # @return [TextMapInjector]
+        def initialize(default_setter = Context::Propagation.text_map_setter)
+          @default_setter = default_setter
+        end
+
+        # Set the span context on the supplied carrier.
+        #
+        # @param [Context] context The active Context.
+        # @param [optional Callable] setter An optional callable that takes a
+        #   carrier and a key and a value and assigns the key-value pair in the
+        #   carrier. If omitted the default setter will be used which expects
+        #   the carrier to respond to [] and []=.
+        # @yield [Carrier, String, String] if an optional setter is provided,
```

Comment:
Docs are out of date here too. The setter is no longer a block. Take a look at the docs for other injectors.